src/erl.rs

```diff
@@ -142,30 +143,108 @@ pub fn record_definition(name: &str, fields: &[&str]) -> String {
 
 pub fn module(module: &TypedModule, writer: &mut impl Utf8Writer) -> Result<()> {
     let module_name = module.name.as_slice();
-    let exports = concat(
-        module
-            .statements
-            .iter()
-            .flat_map(|s| match s {
-                Statement::Fn {
-                    public: true,
-                    name,
-                    args,
-                    ..
-                } => Some((name.clone(), args.len())),
+    let mut exports = vec![];
+    let mut type_defs = vec![];
+    let mut type_exports = vec![];
+    for s in &module.statements {
+        match s {
+            Statement::Fn {
+                public: true,
+                name,
+                args,
+                ..
+            } => exports.push(atom(name.to_string()).append("/").append(args.len())),
+            Statement::ExternalFn {
+                public: true,
+                name,
+                args,
+                ..
+            } => exports.push(atom(name.to_string()).append("/").append(args.len())),
+            Statement::CustomType {
+                name,
+                parameters,
+                public,
+                constructors,
+                opaque,
+                ..
+            } => {
+                // Type Exports
+                if *public {
+                    type_exports.push(
+                        name.to_snake_case()
+                            .to_doc()
+                            .append("/")
+                            .append(parameters.len()),
+                    )
+                }
+                // Type definitions
+                let constructors = concat(
+                    constructors
+                        .iter()
+                        .map(|c| {
+                            // TODO more constructor stuff
+                            // {name, values, values, ...}
+                            let name = c.name.to_snake_case().to_doc();
+                            let constructor = if c.args.is_empty() {
+                                name
+                            } else {
+                                name.append(", ").append(concat(
+                                    c.args
+                                        .iter()
+                                        .map(|a| a.1.to_erlang_type_def())
+                                        .intersperse(", ".to_doc()),
+                                ))
+                            };
+                            "{".to_doc().append(constructor).append("}")
+                        })
+                        .intersperse(" | ".to_doc()),
```

Comment:
It would be nice to wrap and indent definitions if they get too long. Doesn't have to be super pretty (the generated Erlang is a bit ugly) but something like this would be good:

```erl
-type my_type() :: the_first_atom |
  the_second_atom |
  the_third_atom |
  the_forth_atom |
  the_fifth_atom.
```

Something like this perhaps:

```rust
concat(constructors
    .iter()
    .map(constructor_type_definition)
    .intersperse(break_("| ", " | ")))
    .nest(INDENT)
    .group()
```

Something similar for the type tuples would also be good. I think there is a tuple function that would work.